Plugins/Player/Player.cpp

```diff
@@ -1336,4 +1344,163 @@ ArgumentStack Player::SetResManOverride(ArgumentStack&& args)
+                    
+                    if (entries.num > 0)
+                    {
+                        auto pEntry = entries.element;
+                        for (i = 0; i < entries.num; i++, pEntry++)
+                        {
+                            if (pEntry->szPlot_Id.CStr() == tag)
+                            {
+                                pCreature->m_pJournal->m_lstEntries[i] = newJournal;
+                                break;
+                            }
+                        }
+                    }
+                    if(i == entries.num)
+                        pCreature->m_pJournal->m_lstEntries.Add(newJournal);
+                    
+                    pMessage->SendServerToPlayerJournalAddQuest(pPlayer,
+                                                                    newJournal.szPlot_Id,
+                                                                    newJournal.nState,
+                                                                    newJournal.nPriority,
+                                                                    newJournal.nPictureIndex,
+                                                                    newJournal.bQuestCompleted,
+                                                                    newJournal.nCalendarDay,
+                                                                    newJournal.nTimeOfDay,
+                                                                    newJournal.szName,
+                                                                    newJournal.szText);
+                    retval =pCreature->m_pJournal->m_lstEntries.num; // Success
+                    
+                    //If no update message is desired, we can keep it silent.
+                    //Useful for restoring entries on playerlogin
+                    if(!silentUpdate)
+                    {
+                        pMessage->SendServerToPlayerJournalUpdated(pPlayer,1,newJournal.bQuestCompleted,newJournal.szName);
+                    }
+                }
+                else
+                {
+                    LOG_ERROR("Unable to get CNWSMessage");
+                }
+                
+        }
+    }
+
+    return Services::Events::Arguments(retval);
+}
+
+ArgumentStack Player::GetJournalEntry(ArgumentStack&& args)
+{
+    SJournalEntry lastJournalEntry;
+    int found = -1;
+    if (auto *pPlayer = player(args))
+    {
+        auto *pCreature = Globals::AppManager()->m_pServerExoApp->GetCreatureByGameObjectID(pPlayer->m_oidNWSObject);
+        if (pCreature && pCreature->m_pJournal)
+        {
+            auto entries = pCreature->m_pJournal->m_lstEntries;
+            const auto tag = Services::Events::ExtractArgument<std::string>(args);
+            ASSERT_OR_THROW(tag.size() > 0);
+            if (entries.num > 0)
+            {
+                auto pEntry = entries.element;
+                for (int i = 0; i < entries.num; i++, pEntry++)
```

Comment:
Please no pointer arithmetic :)

`CExoArrayList<>` actually supports modern C++ iterators, so you can probably do:
```C
for (auto& entry : entries)
{
  ...
}
```